src/main/java/core/net/OnlineWorker.java

```diff
@@ -302,19 +328,27 @@ public static boolean downloadMatchData(MatchKurzInfo info, boolean refresh) {
 							}
 						}
 
-						// get the other team
-						int otherId;
-						if (info.isHomeMatch()) {
-							otherId = info.getGuestTeamID();
-						} else {
-							otherId = info.getHomeTeamID();
-						}
-						if (otherId > 0) {
-							Map<String, String> otherTeam = getTeam(otherId);
-							info.setIsDerby(getRegionId(otherTeam) == HOVerwaltung.instance().getModel().getBasics().getRegionId());
-							info.setIsNeutral(info.getArenaId() != HOVerwaltung.instance().getModel().getStadium().getArenaId()
-									&& info.getArenaId() != getArenaId(otherTeam));
-							downloadTeamLogo(otherTeam);
+						if (info.getHomeTeamID() > 0) {
+							int guestRegionId;
+							if (info.isAwayMatch()) {
+								guestRegionId = HOVerwaltung.instance().getModel().getBasics().getRegionId();
+							} else {
+								var guestTeamInfo = getTeam(info.getGuestTeamID());
+								guestRegionId = getRegionId(guestTeamInfo);
+								downloadTeamLogo(guestTeamInfo);
+							}
+							info.setIsDerby(guestRegionId == info.getRegionId());
+
+							int homeArenaId;
+							if(info.isHomeMatch()){
```

Comment:
I suggest using the default code formatter from IDEA.

Reply:
changed